src/modules/prompt-finder/hole-patching.ts

```diff
@@ -0,0 +1,169 @@
+    templateValue: PromptTemplateHole
+): Promise<JSONSchemaObject> {
+    let userPromptToSend = HoleFillingPromptJson.user_prompt;
+    let systemPromptToSend = HoleFillingPromptJson.system_prompt;
+    let promptWithHoles = prompt.normalizedText;
+    // fill in the holes in the prompt where default values are ready : this should make the default values more consistent with each other
+    for (let key in prompt.templateValues) {
+        let value = prompt.templateValues[key].defaultValue;
+        if (value) {
+            promptWithHoles = promptWithHoles.replace('{{' + key + '}}', value);
+        }
+    }
+    userPromptToSend = userPromptToSend.replace(
+        '{{' + HoleFillingPromptJson.injected_variables[0] + '}}',
+        promptWithHoles
+    );
+    let variableName = templateValue.name;
+    userPromptToSend = userPromptToSend.replace(
+        '{{' + HoleFillingPromptJson.injected_variables[1] + '}}',
+        variableName
+    );
+    // load source code file contents from file
+    const sourceCodeFileContents = fs.readFileSync(
+        prompt.sourceFilePath,
+        'utf8'
+    );
+
+    // inject source code file contents into the prompt
+    userPromptToSend = userPromptToSend.replace(
+        '{{' + HoleFillingPromptJson.injected_variables[2] + '}}',
+        sourceCodeFileContents
+    );
+    // look for read me file in the same directory , or the repository root
+    let readmeFilePath = findReadmeFile(prompt.sourceFilePath);
+    let readmeFileContents = '';
+    if (readmeFilePath) {
+        readmeFileContents = await fs.promises.readFile(readmeFilePath, 'utf8');
+        // inject readme file contents into the prompt
+        userPromptToSend +=
+            '\n You may use the following README.md file contents to help you better understand the context of this prompt: "\n' +
+            readmeFileContents +
+            '"';
+    }
+
+    // const deploymentId = 'gpt-35-turbo';
+    const messages: ChatCompletionMessageParam[] = [
+        { role: 'system', content: systemPromptToSend },
+        { role: 'user', content: userPromptToSend },
+    ];
+    let client = utils.getClient();
+    // console.log(client);
+    if (client === undefined) {
+        console.error('Client is undefined');
+        return JSON.parse('{"error": " Issue during OpenAI configuration"}');
+    } else {
+        try {
+            // const response = await client.chat.completions.create({
+            //     messages: messages,
+            //     model: deploymentId,
+            //     temperature: 0.3,
+            //     seed: 42,
+            // });
+            // console.log(response);
+            // const result = response.choices?.[0]?.message?.content;
+            const result = await utils.sendChatRequest(messages, {
+                model: modelType,
+                temperature: 0.3,
+                seed: 42,
+                // type: "json_object" // force answer to be valid json ==> NOT SUPPORTED BY AZURE
```

Comment:
@dhia-e-rzig i think you said we swapped off azure, is it safe to turn this back on?

Reply:
I'd have to check docs, but hopefully AzOAI endpoint supports this param soon! (Maybe w/ an `api_version=...` update)